ui/src/shared/components/ScatterContainer.tsx

```diff
@@ -50,9 +57,13 @@ const ScatterContainer: FunctionComponent<Props> = ({
     colors,
     xDomain: storedXDomain,
     yDomain: storedYDomain,
+    xColumn,
+    yColumn,
   },
   onSetFillColumns,
   onSetSymbolColumns,
+  onSetXColumn,
+  onSetYColumn,
 }) => {
   useEffect(() => {
     if (fluxGroupKeyUnion && (!storedSymbol || !storedFill)) {
```

Comment:
What is an instance where there is no `storedSymbol` / `storedFill` for a scatter view? I think this should never happen... or if it is happening, then the fix should be elsewhere so that we can ensure it never happens rather than programming defensively here. Can we ensure that the `storedSymbol` and `storedFill` settings are valid when the view is created (in the VEO)?

I think this `useEffect` introduces two-way data flow (component accepting data and emitting effects to change that data) into a component that should only observe one way data flow (component only accepts data). It feels like we are fighting React a lot here.